### DISCUSSION TITLE
Should we release 0.71.11?

### DISCUSSION THREAD

**Discussion by kelset** [reactwg/react-native-releases#70] [Patches]
## Released, please refer to #73

Current Release: [0.71.10](https://github.com/facebook/react-native/releases/tag/v0.71.10)

Conversations on this thread are limited:

- [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking).
- [qualified cherry-pick requests](https://reactnative.dev/contributing/release-faq#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://reactnative.dev/contributing/release-faq#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).

Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking), please reference the issue here.

---

#### List of qualified picks

1. revert https://github.com/facebook/react-native/commit/cd6a91343ee24af83c7437b3f2449b41e97760e9
1. ~~https://github.com/facebook/react-native/commit/3681df287835f5467a2ad5afe950eae16b95fd8b~~
1. https://github.com/facebook/react-native/commit/f7e35d4ef7d68d06fba1439c0aa6d9ed05b58a7f
1. https://github.com/facebook/react-native/commit/c05d822f7daa92e8af2ec2cd97a9897425624cc2
1. https://github.com/facebook/react-native/commit/671ea383fe45dd9834a0c0481360de050df7f0c9
1. https://github.com/facebook/react-native/commit/10d55888cce0b625c4b05a19ed609154893d17d0
1. bump: Metro 0.73.10 & CLI 10.2.4

#### Local commits to backport to main

1.

**Comment by cipolleschi** [comment]
We will need this for Xcode15 
- https://github.com/facebook/react-native/pull/37758

**Comment by fluiddot** [comment]
It would be great if we could backport https://github.com/facebook/react-native/commit/f7e35d4ef7d68d06fba1439c0aa6d9ed05b58a7f (https://github.com/facebook/react-native/pull/36121) to version `0.71`.

**Comment by robhogan** [comment]
Nominate this one for the RN-side fix to https://github.com/facebook/react-native/issues/36794 (to accompany a bump to Metro 0.73.10).
 - https://github.com/facebook/react-native/pull/37501 (https://github.com/facebook/react-native/commit/671ea383fe45dd9834a0c0481360de050df7f0c9)

**Reply by kelset** [reply]
bump to do:
* Metro 0.73.10
* CLI 10.2.4 https://github.com/react-native-community/cli/commits/10.x

**Comment by AlexanderEggers** [comment]
Would it be possible to backport https://github.com/facebook/react-native/commit/3681df287835f5467a2ad5afe950eae16b95fd8b? That fixes certain layout flatting issues. This issue is already fixed in 0.72 but it would be great to have that in 0.71 already.

**Reply by LcTwisk** [reply]
Just linking the layout flattening issue for anyone interested: https://github.com/facebook/react-native/issues/37797

**Reply by kelset** [reply]
we tried to cherry pick it but it created significant merge conflicts so we are backing out to make sure we won't break things further. You can try submitting a PR yourself that does the cherry pick directly against the 0.71 branch so that we can review it and merge it for next patch (0.71.12).

**Reply by AlexanderEggers** [reply]
@kelset I created a cherry-pick here: https://github.com/facebook/react-native/pull/37876

**Comment by davidgvf** [comment]
Hello, i have a problem in build IOS, not found RCTAppDelegate
Does anyone know something? I use firebase crashlytics and my podfile is this:

require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
$RNFirebaseAsStaticFramework = true
$RNFirebaseAnalyticsWithoutAdIdSupport = true

platform :ios, min_ios_version_supported
use_frameworks! :linkage => :static
use_modular_headers!
prepare_react_native_project!

# If you are using a `react-native-flipper` your iOS build will fail when `NO_FLIPPER=1` is set.
# because `react-native-flipper` depends on (FlipperKit,...) that will be excluded
#
# To fix this you can also exclude `react-native-flipper` using a `react-native.config.js`
# ```js
# module.exports = {
#   dependencies: {
#     ...(process.env.NO_FLIPPER ? { 'react-native-flipper': { platforms: { ios: null } } } : {}),
# ```
flipper_config = ENV['NO_FLIPPER'] == "1" ? FlipperConfiguration.disabled : FlipperConfiguration.enabled

linkage = ENV['USE_FRAMEWORKS']
if linkage != nil
  Pod::UI.puts "Configuring Pod with #{linkage}ally linked Frameworks".green
  use_frameworks! :linkage => linkage.to_sym
end

target 'KubikApp' do
  config = use_native_modules!

  # Flags change depending on the env values.
  flags = get_default_flags()

  use_react_native!(
    :path => config[:reactNativePath],
    :hermes_enabled => flags[:hermes_enabled],
    :fabric_enabled => flags[:fabric_enabled],
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )

  target 'KubikAppTests' do
    inherit! :complete
    # Pods for testing
  end

  post_install do |installer|
    react_native_post_install(
      installer,
      # Set `mac_catalyst_enabled` to `true` in order to apply patches
      # necessary for Mac Catalyst builds
      :mac_catalyst_enabled => false
    )
    __apply_Xcode_12_5_M1_post_install_workaround(installer)
  end
end

**Comment by onmotion** [comment]
Could we please backport https://github.com/facebook/react-native/commit/c05d822f7daa92e8af2ec2cd97a9897425624cc2 into release

**Comment by tido64** [comment]
Xcode defaults to 'Release' when running `react-native run-ios` under specific setups. This is fixed in `@react-native-community/cli` 10.2.4: https://github.com/react-native-community/cli/pull/1957#issuecomment-1587197439

**Comment by AlexanderEggers** [comment]
Would it be possible to revert https://github.com/facebook/react-native/commit/2558c3d4f56776699602b116aff8c22b8bfa176a which was introduced in the last RN patch version? This is causing quite a bit of side-effects with reanimated which needs to be patched manually. I would argue that such breaking impact should not be introduced in a patch release but should be reserved for larger package updates. Ref: https://github.com/software-mansion/react-native-reanimated/issues/4548

**Reply by kelset** [reply]
this is fixed in reanimated 3.3.0? https://github.com/software-mansion/react-native-reanimated/pull/4550

It doesn't sound like we did break anything, that commit is just a types fix for https://github.com/facebook/react-native/issues/37543

**Reply by AlexanderEggers** [reply]
Yep that is correct but not every project can update to reanimated v3 yet.

**Comment by kelset** [comment]
note to self: we need to see if we can backport this https://github.com/facebook/react-native/pull/37840 once it's merged, or manually revert https://github.com/facebook/react-native/commit/cd6a91343ee24af83c7437b3f2449b41e97760e9

**Comment by NickGerleman** [comment]
It would be great to backport https://github.com/facebook/react-native/commit/3681df287835f5467a2ad5afe950eae16b95fd8b to fix https://github.com/facebook/react-native/issues/37797.

**Reply by kelset** [reply]
was already asked here btw https://github.com/reactwg/react-native-releases/discussions/70#discussioncomment-6133285

**Reply by kelset** [reply]
replying here too: we tried to cherry pick it but it created significant merge conflicts so we are backing out to make sure we won't break things further. You can try submitting a PR yourself that does the cherry pick directly against the 0.71 branch so that we can review it and merge it for next patch (0.71.12).
